experimental/Schemes/NormalToricVarieties/attributes.jl

```diff
@@ -51,29 +51,119 @@
 # too. On the other hand, none of the vⱼ was in τ⟂ for j > s, 
 # so neither can be any convex combination, but the trivial one. 
 # This proves the claim. Now (*) follows directly.
-function _torusinvariant_weil_divisors(X::NormalToricVariety; check::Bool=false)
-  if has_attribute(X, :_torusinvariant_weil_divisors)
-    return get_attribute(X, :_torusinvariant_weil_divisors)::Vector{<:AbsWeilDivisor}
-  end
-  ray_list = rays(polyhedral_fan(X))
-  ideal_sheaves = Vector{IdealSheaf}()
-  for tau in ray_list
-    tau_dual = polarize(cone(tau))
-    ideal_dict = IdDict{AbsSpec, Ideal}()
-    for U in affine_charts(X)
-      if !(tau in cone(U))
-        ideal_dict[U] = ideal(OO(U), one(OO(U)))
-        continue
+function _torusinvariant_weil_divisors(X::NormalToricVariety; check::Bool=false, algorithm::Symbol=:via_polymake)
+  return get_attribute!(X, :_torusinvariant_weil_divisors) do
+    ray_list = rays(polyhedral_fan(X))
```

Comment:
I understand this function works as long as `X` is simplicial? So maybe:
```suggestion
    @req is_simplicial(X) "Variety must be simplicial"
    ray_list = rays(polyhedral_fan(X))
```

Reply:
I think @lkastner is the right person to ask. If this applies, we should add an `@check`.